.circleci/ci/lighthouse.js

```diff
@@ -0,0 +1,161 @@
+/*******************************************************************************
+ * Copyright 2023 Adobe
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ ******************************************************************************/
+
+const fs = require('fs');
+const https = require('https');
+const ci = new (require('./ci.js'))();
+
+
+const checkLightHouse = async () => {
+    const lighthouse = await import('lighthouse')
+    const chromeLauncher = await import('chrome-launcher')
+    const chrome = await chromeLauncher.launch({chromeFlags: ['--headless']});
+    const options = {logLevel: 'info', output: 'html', port: chrome.port, extraHeaders: { Authorization: 'Basic ' + Buffer.from('admin:admin').toString('base64') }};  // YWRtaW46YWRtaW4= -- base64 encoded, admin:admin
+
+    const lighthouseConfig = JSON.parse(fs.readFileSync('/home/circleci/build/.circleci/ci/lighthouseConfig.json'))
+
+    const runnerResult = await lighthouse.default(lighthouseConfig.urls[0], options);
+    // `.report` is the HTML report as a string
+
+    const reportHtml = runnerResult.report;
+    console.log('Report is done for', runnerResult.lhr.finalDisplayedUrl);
+    console.log(getCommentText(runnerResult.lhr.categories))
+
+    //postCommentToGitHub(getCommentText(runnerResult.lhr.categories), process.env.GITHUB_TOKEN)
+    ci.sh('mkdir artifacts');
+    ci.dir("artifacts", () => {
+           fs.writeFileSync('LigthouseReport.html', reportHtml);
+        });
+
+
+    const thresholdResults = checkThresholds(runnerResult.lhr.categories, lighthouseConfig)
+    console.log("thresholdResults -->>>> ", thresholdResults)
+    if(!thresholdResults.isThresholdPass){
+        console.log("Error: Lighthouse score for aem-core-forms-components, below the thresholds")
+        process.exit(1);
+    }
+    else if(thresholdResults.updateLighthouseConfig && process.env.CIRCLE_BRANCH == 'master'){ // only execute if branch name is 'master'
```

Comment:
Please fix the branch name check. Since we also other release branches

Reply:
added 'dev' and 'master'